File: lib/domain/cultivation_plan/dtos/add_crop_adjust_result.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Dtos
      # plan_allocation_adjust サブステップの結果（add_crop オーケストレータが読む）。
      class AddCropAdjustResult
        attr_reader :success, :message, :http_status

        def initialize(success:, message: nil, http_status: nil)
          @success = !!success
          @message = message
          @http_status = http_status
          freeze
        end

        def success?
          @success
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/dtos/add_crop_adjust_result.rs
//! Ruby: `Domain::CultivationPlan::Dtos::AddCropAdjustResult`

#[derive(Debug, Clone, PartialEq)]
pub struct AddCropAdjustResult {
    pub success: bool,
    pub message: Option<String>,
    pub http_status: Option<i32>,
}

impl AddCropAdjustResult {
    pub fn success() -> Self {
        Self {
            success: true,
            message: None,
            http_status: None,
        }
    }

    pub fn failure(message: impl Into<String>, http_status: Option<i32>) -> Self {
        Self {
            success: false,
            message: Some(message.into()),
            http_status,
        }
    }

    pub fn is_success(&self) -> bool {
        self.success
    }
}
